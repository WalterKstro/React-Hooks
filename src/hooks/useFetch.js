import { useState, useEffect, useRef } from "react"
import fetchAPI from "../helpers/fetchAPI"

const useFetch = (url) => {
    const [state, setState] = useState({data: null,loading: true})
    const isMounted = useRef(true)

    useEffect(() => {
        return ()=> {
            isMounted.current = false
        }
    },[])
    
    useEffect(() => {
    
        fetchAPI(url).then(data => {
            if(isMounted.current) {
                setState({
                    data,
                    loading: false
                })            
            }
        })

    },[url])


    return state
}

export default useFetch
