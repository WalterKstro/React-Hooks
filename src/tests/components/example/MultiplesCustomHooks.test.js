import { shallow } from "enzyme"
import MultiplesCustomHooks from "../../../components/examples/MultiplesCustomHooks"
import useFetch from "../../../hooks/useFetch"
jest.mock('../../../hooks/useFetch')

describe('Testing multiple custom hooks in component <MultiplesCustomHooks/>', () => {

    test('should mach with the snapshot', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true
        })
        const renderedComponent = shallow(<MultiplesCustomHooks/>)
        expect(renderedComponent).toMatchSnapshot()  
    })

    test('should show data of custom hook useFetch', () => {
        const data = {
            quote: "IFT",
            author: "Skyler White",
            series: "Breaking Bad"
        }
        useFetch.mockReturnValue({
            data: [data],
            loading: false
        })
        const renderedComponent = shallow(<MultiplesCustomHooks/>)
        const isQuote = renderedComponent.exists('Quote')
        const isLoading = renderedComponent.exists('LoadingWait')
        
        expect(isQuote).toBe(true)
        expect(isLoading).toBe(false)
        

    })
    
    
})