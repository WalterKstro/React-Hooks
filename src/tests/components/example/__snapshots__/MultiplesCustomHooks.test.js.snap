// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing multiple custom hooks in component <MultiplesCustomHooks/> should mach with the snapshot 1`] = `
<div
  className="p-3"
>
  <h2
    className="my-3 text-center"
  >
    Breaking Bad
  </h2>
  <LoadingWait />
  <div
    className="d-grid gap-2 d-md-flex justify-content-md-end"
  >
    <button
      className="font-weight btn btn-primary me-md-2"
      onClick={[Function]}
      type="button"
    >
      «
    </button>
    <button
      className="font-weight btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      »
    </button>
  </div>
</div>
`;
